contracts/FeralfileArtworkV4.sol

```diff
@@ -141,9 +118,8 @@ contract FeralfileExhibitionV4 is
         burnable = burnable_;
         bridgeable = bridgeable_;
         costReceiver = costReceiver_;
-        vault = vault_;
+        vault = FeralfileVault(payable(vault_));
```

Comment:
why not just the address and need to make the whole instance public?

Reply:
I'm thinking that init a instance everytime in buyArtwork seems wasted. we could do this once while contract init.
But I don't do the test if the init instance cost much, maybe it doesn't cost any :P
my thought is it just no harm to put it at constructor level.

Reply:
If so, I think we make the `vault` a private variable and create another variable in public variable just for address.

I do not know what people will see in etherscan, but if vault is an object, it looks like can not get help information when we check from the etherscan.